js/models/conversations.js

```diff
@@ -217,9 +217,10 @@
       this.messageCollection.forEach(m => m.trigger('change'));
     },
 
-    bumpTyping() {
+    async bumpTyping() {
       // We don't send typing messages if the setting is disabled or we aren't friends
-      if (!this.isFriend() || !storage.get('typing-indicators-setting')) {
+      const hasFriendDevice = await this.isFriendWithAnyDevice();
+      if (!storage.get('typing-indicators-setting') || !hasFriendDevice) {
```

Comment:
Does this mean we can send typing messages to devices for which there is no session yet?
I'm trying to see the implications...

Reply:
Sorry one of those "does make sense without the rest of the branch" things
The plan is for any outgoing message, we try and send it to all the devices
If we are friends with the device then just send the message to that device, if we aren't friends with that device then we will check the timestamp of the last friend request sent to that device and if it has been >= 4 days for example then we will send an automated background friend request to that device

Reply:
So what is currently happening in the rest of my branch is that every outgoing message is either sent normally or replaced with an automated friend request

Reply:
Ok that makes sense.